examples/cookbook/recipe_shared_memory.cpp

```diff
@@ -0,0 +1,120 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/config.hpp"
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/resource/HostSharedMemoryResource.hpp"
+#include "umpire/util/MemoryResourceTraits.hpp"
+
+#include "mpi.h"
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <thread>
+
+//
+// For debugging purposes, this program uses the number of command line
+// as a flag to indicate the mode it should run in.  The modes are:
+//
+// 1) If run with no argument (ac==1), it will run as an MPI program.
+// 2) If run with 1 argument (ac==2), it will run as the parent non-mpi program.
+// 3) If run with 2 arguments (ac==3), it will run as the child non-mpi program.
+//
+// This will allow someone to launch this program as a Parent and Child in two
+// separate debugger session windows (possible in gdb or vscode).  When running
+// in the debugger session windows, no MPI will be used and the debugger must
+// be used for synchronization (by setting breakpoints).
+//
+int main(
+  int
+  ac
+  ,
+  char**
+  av
+)
+{
+  const bool use_mpi{ ac == 1 };
+  const bool i_am_parent{ ac == 2 };
+
+  if (use_mpi) {
+    MPI_Init(&ac, &av);
+  }
+
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  //
+  // Set up the traits for the allocator
+  //
+  auto traits{umpire::get_default_resource_traits("SHARED")};
+  traits.size = 1*1024*1024;  // Maximum size of this Allocator
```

Comment:
For this whole example, it seems like it goes back and forth between initializing like "int var = 0;" and "int var {0};" - should we keep this more consistent?

Reply:
Not sure I follow.  Do you have a suggestion for this line?

Reply:
Sorry, Marty, I misread this - it's fine! Whoopsies.